fn_jira/fn_jira/components/jira_common.py

```diff
@@ -0,0 +1,208 @@
+import requests
+import json
+import logging
+import re
+from resilient_circuits import FunctionError
+
+'''
+This module implements the calls needed for jira api access. API operations supported:
+) create an issue
+) create a comment
+) transition an issue
+'''
+
+# URL fragments needed along with the base jira URL
+SEARCH_URL = 'rest/api/2/search'
+ISSUE_URL  = 'rest/api/2/issue'
+TRANSITION_PARAM = 'transitions'
+
+COMMENT_PARAM = 'comment'
+
+def create_issue(log, appDict):
+    """Function: create a jira issue.
+    Return the raw JSON returned from the api call
+    """
+
+    issue_url = '/'.join((appDict['url'], ISSUE_URL))
+
+    payload = _mkCreatePayload(appDict)
+    log.info(payload)
+
+    result = execute_call(log, 'post', issue_url, appDict['user'], appDict['password'], payload)
+    # log.info(result)
+
+    return result
+
+
+def transition_issue(log, appDict):
+    """Function: transition a jira issue.
+    Return the raw JSON returned from the api call
+    """
+
+    url = '/'.join((appDict['url'], TRANSITION_PARAM))
+    payload = _mkTransitionPayload(appDict)
+    log.info(payload)
+
+    result = execute_call(log, 'post', url, appDict['user'], appDict['password'], payload)
+    # log.info(result)
+
+    if appDict['comment']:
+        create_comment(log, appDict)
+
+    return result
+
+def create_comment(log, appDict):
+    """Function: create a jira comment in a Jira issue. No JSON is returned on success"""
+
+    url = '/'.join((appDict['url'], COMMENT_PARAM))
+
+    payload = _mkCommentPayload(appDict)
+    log.info(payload)
+
+    resp =  execute_call(log, 'post', url, appDict['user'], appDict['password'], payload)
+
+    return resp
+
+
+def execute_call(log, verb, url, user, password, payload):
+    """Function: perform the http API call. Different types of http operations are supported: 
+        GET, POST, PUT
+        Errors raise FunctionError
+    """
+    
+    try:
+        headers = {'content-type': 'application/json'}
+
+        log.info(payload)
+
+        if verb == 'get':
+            resp = requests.get(url, verify=False, headers=headers, data=payload, auth=(user, password)) ## verify=False for debugging
```

Comment:
Instead of hard coded verify=False here, is it better to use a function parameter for this? Then customer can use workflow input to control this.